trade-finance-manager-ui/server/controllers/helpers/can-deal-be-cancelled.helper.ts

```diff
@@ -16,14 +23,19 @@ export const canSubmissionTypeBeCancelled = (submissionType: DealSubmissionType)
  * Checks if deal cancellation is enabled for a deal and user type
  * @param submissionType - the deal submission type
  * @param user - the session user
+ * @param cancellationStatus - the current deal cancellation status if it exists
  * @returns true or false depending on the feature flag, submission type and user type
  */
-export const isDealCancellationEnabled = (submissionType: DealSubmissionType, user: TfmSessionUser): boolean => {
-  // TODO: DTFS2-7298: also check that the deal hasn't already been cancelled.
+export const canDealBeCancelled = (submissionType: DealSubmissionType, user: TfmSessionUser, cancellationStatus?: TfmDealCancellationStatus): boolean => {
+  const isDealCancellationFeatureFlagEnabled = isTfmDealCancellationFeatureFlagEnabled();
 
   const isUserAllowedToCancelDeal = userIsInTeam(user, [TEAM_IDS.PIM]);
-  const isDealCancellationFeatureFlagEnabled = isTfmDealCancellationFeatureFlagEnabled();
+
   const isAcceptableSubmissionType = canSubmissionTypeBeCancelled(submissionType);
 
-  return isUserAllowedToCancelDeal && isDealCancellationFeatureFlagEnabled && isAcceptableSubmissionType;
+  const isDealNotAlreadyCancelled = !(
```

Comment:
Finding `isDealNotAlreadyCancelled` a little confusing especially in the final return, how about something like this?

```ts
const dealIsAlreadyCancelled = Object.values(TFM_DEAL_CANCELLATION_STATUS).includes(cancellationStatus);
```

Actually `Object.values` probably won't work here, maybe just remove the `!` and rename the existing const?

Then the return can do `!dealIsAlreadyCancelled`

Reply:
Agree `dealIsAlreadyCancelled` is much easier to read :) And since we're negating here anyway it's probably simpler to see whats going on anyway

Reply:
Have changed! Thanks both